frontend/src/Components/ScaleSelector/ScaleSelector.tsx

```diff
@@ -7,122 +8,143 @@ import classes from './ScaleSelector.module.css';
-        onBlur={handleBlur}
-        aria-expanded={open}
-        aria-haspopup="listbox"
-      >
-        {SELECT_CHANGE_SCALE}
-      </button>
-      {open ? (
-        <ul
-          class={classes.dropDown}
-          ref={dropdownRef}
-          tabIndex={0}
-          role="listbox"
-          aria-label="scales"
+    useEffect(() => {
+      if (!open) {
+        return;
+      }
+      if (!connected) {
+        close();
+        return;
+      }
+      if (dropdownRef.current) {
+        dropdownRef.current.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+        dropdownRef.current.focus();
+      }
+    }, [open, connected]);
+
+    return (
+      <div class={classes.scaleSelector}>
+        <button
+          disabled={!connected}
+          onClick={toggle}
+          class={sharedClasses.button}
+          ref={selectionButtonRef}
           onBlur={handleBlur}
-          onKeyDown={handleKeyboardNavigation}
+          aria-expanded={open}
+          aria-haspopup="listbox"
         >
-          {availableScales.map(([id, { name }], index) => (
-            <li
-              class={classes.dropDownItem}
-              key={id}
-              role="option"
-              aria-selected={selected === index}
-              onClick={() => {
-                close();
-                setScale(SCALES[id as ScaleName].values);
-              }}
-              onMouseMove={() => selected !== index && setSelected(index)}
-              onMouseLeave={() => selected === index && setSelected(-1)}
-            >
-              {name}
-            </li>
-          ))}
-        </ul>
-      ) : null}
-    </div>
-  );
-});
+          {SELECT_CHANGE_SCALE}
+        </button>
+        {open ? (
+          <ul
+            class={classes.dropDown}
+            ref={dropdownRef}
+            tabIndex={0}
+            role="listbox"
+            aria-label="scales"
+            onBlur={handleBlur}
+            onKeyDown={handleKeyboardNavigation}
+          >
+            {availableScales.map(([id, { name }], index) => (
+              <li
+                class={classes.dropDownItem}
+                key={id}
+                role="option"
+                aria-selected={selected === index}
+                onClick={() => {
+                  close();
+                  setScale(SCALES[id as ScaleName].values);
+                }}
+                onMouseMove={() => selected !== index && setSelected(index)}
+                onMouseLeave={() => selected === index && setSelected(-1)}
+              >
+                {name}
+              </li>
+            ))}
+          </ul>
+        ) : null}
+      </div>
+    );
+  }
+);
+
+// Unfortunately, the % operator in JavaScript is not a true modulo operator.
+// The helper below works for arbitrary dividends even though >= -1 would be sufficient
+const unsignedModulo = (dividend: number, divisor: number) =>
+  ((dividend % divisor) + divisor) % divisor;
+
+const isScaleSelected =
```

Comment:
Since this is a second order function that returns a function that might be named `isScaleSelected`, I would prefer a name a long the line of
```suggestion
const getScaleMatcher =
```

Reply:
Done